components/admin-panel/sections/ActivityLog/ActivityDescription.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { capitalize } from 'lodash';
+import { useIntl } from 'react-intl';
+
+import { ActivityDescriptionI18n } from '../../../../lib/i18n/activities';
+import formatMemberRole from '../../../../lib/i18n/member-role';
+import { getCollectivePageRoute } from '../../../../lib/url-helpers';
+
+import Link from '../../../Link';
+import LinkCollective from '../../../LinkCollective';
+import LinkExpense from '../../../LinkExpense';
+
+const ActivityDescription = ({ activity }) => {
+  const intl = useIntl();
+
+  if (!ActivityDescriptionI18n[activity.type]) {
+    return capitalize(activity.type.replace('_', ' '));
+  }
+
+  return intl.formatMessage(ActivityDescriptionI18n[activity.type], {
+    FromAccount: () => <LinkCollective collective={activity.fromAccount} openInNewTab />,
+    Account: () => <LinkCollective collective={activity.account} openInNewTab />,
+    Expense: msg =>
+      !activity.expense ? (
+        msg
+      ) : (
+        <LinkExpense
+          collective={activity.expense.account}
+          expense={activity.expense}
+          title={activity.expense.description}
+          openInNewTab
+        >
+          {msg} #{activity.expense.legacyId}
+        </LinkExpense>
+      ),
+    Order: msg =>
+      !activity.order ? (
+        msg
+      ) : (
+        <Link
+          href={`${getCollectivePageRoute(activity.order.toAccount)}/orders?searchTerm=%23${activity.order.legacyId}`}
+          title={activity.order.description}
+          openInNewTab
+        >
+          {msg} #{activity.order.legacyId}
+        </Link>
+      ),
+    Host: () => <LinkCollective collective={activity.host} openInNewTab />,
```

Comment:
Same as my comment above to add `title` prop and set it to collective description if that exists. If it doesn't we use `noTitle` prop.